src/paths/johnson.c

```diff
@@ -91,7 +94,7 @@ int igraph_shortest_paths_johnson(const igraph_t *graph,
     }
 
     /* If no negative weights, then we can run Dijkstra's algorithm */
-    if (no_of_edges > 0 && igraph_vector_min(weights) >= 0) {
+    if (no_of_edges == 0 || igraph_vector_min(weights) >= 0) {
```

Comment:
Could we combine the `igraph_vector_min` call with NaN detection, and get rid of the preceding `igraph_vector_is_any_nan` call?  `igraph_vector_min` returns NaN if any elements are NaN.

As usual, we must continue to ensure that vector_min is not called for a zero-length vector.

Reply:
I will do this and merge.